--- voltdb.php.old	2010-08-25 11:01:33.807770905 -0400
+++ voltdb.php	2010-08-25 11:03:42.186093443 -0400
@@ -2244,7 +2244,7 @@
 
 	public function invoke($proc,$callback=null) {
 		switch (func_num_args()) {
-		case 1: ClientNative_invoke($this->_cPtr,$proc); break;
+		case 1: return new InvocationResponse(ClientNative_invoke($this->_cPtr,$proc));
 		default: ClientNative_invoke($this->_cPtr,$proc,$callback);
 		}
 	}
@@ -2267,11 +2267,7 @@
 		default: $r=ClientNative_create($listener);
 		}
 		if (is_resource($r)) {
-			$c=substr(get_resource_type($r), (strpos(get_resource_type($r), '__') ? strpos(get_resource_type($r), '__') + 2 : 3));
-			if (!class_exists($c)) {
-				return new ClientNative($r);
-			}
-			return new $c($r);
+			return new ClientNative($r);
 		}
 		return $r;
 	}
@@ -2615,16 +2611,12 @@
 
 	public function acquireClient($hostname,$username,$password,$port_or_listener=null,$port=null) {
 		switch (func_num_args()) {
-		case 3: $this->_cPtr=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password); break;
-		case 4: $this->_cPtr=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password,$port_or_listener); break;
-		default: $this->_cPtr=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password,$port_or_listener,$port);
+		case 3: $r=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password); break;
+		case 4: $r=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password,$port_or_listener); break;
+		default: $r=ConnectionPool_acquireClient($this->_cPtr,$hostname,$username,$password,$port_or_listener,$port);
 		}
 		if (is_resource($r)) {
-			$c=substr(get_resource_type($r), (strpos(get_resource_type($r), '__') ? strpos(get_resource_type($r), '__') + 2 : 3));
-			if (!class_exists($c)) {
-				return new ClientNative($r);
-			}
-			return new $c($r);
+			return new ClientNative($r);
 		}
 		return $r;
 	}
